commit 468b955ec785d8dca94ae00aa982d6d1b0a86410
Author: Torgny Nyblom <nyblom@kde.org>
Date:   Fri Dec 2 09:43:38 2011 +0100

    Make sure that the Bcc header is removed
    
    When sending a mail make sure that the Bcc header is removed before
    sending the mail.
    
    CCBUG: 263587
    REVIEW: 103308
    (cherry picked from commit 0875ca055322fd94962ed473be0a41854e17373a)

diff --git agents/maildispatcher/sendjob.cpp agents/maildispatcher/sendjob.cpp
index 52b0e27..ed02267 100644
--- kdepim-runtime-4.7.4/agents/maildispatcher/sendjob.cpp
+++ kdepim-runtime-4.7.4/agents/maildispatcher/sendjob.cpp
@@ -175,6 +175,10 @@ void SendJob::Private::doTraditionalTransport()
   // Message.
   Q_ASSERT( item.hasPayload<Message::Ptr>() );
   const Message::Ptr message = item.payload<Message::Ptr>();
+  if( message->hasHeader( "Bcc" ) ) {
+    message->removeHeader( "Bcc" );
+    message->assemble();
+  }
   const QByteArray content = message->encodedContent( true ) + "\r\n";
   Q_ASSERT( !content.isEmpty() );
 
